python/paddle/tensor/manipulation.py

```diff
@@ -37,6 +37,128 @@
 __all__ = []
 
 
+@dygraph_only
+def fill_diagonal_tensor_(x, y, dim1=0, dim2=1, offset=0, name=None):
+    """
```

Comment:
Why not unify most of the codes of `fill_diagonal_tensor_` and `fill_diagonal_tensor`. Too many duplicate codes....

Reply:
done